master/buildbot/test/integration/test_upgrade.py

```diff
@@ -135,7 +135,7 @@ def comp(engine):
                     dict(name=idx.name,
                          unique=idx.unique and 1 or 0,
                          column_names=sorted([c.name for c in idx.columns]))
-                    for idx in tbl.indexes])
+                    for idx in tbl.indexes], key=lambda x: x['name'])
```

Comment:
I think we indeed should prefer the lambda x: x['thing'] form. We should probably use that style for all dictionary sort